Add 0.2 changelog (and quick missing spliec doc)

TheBlueMatt opened the pull request:
Note that this is against main so will need to be backported.

valentinewallace commented on CHANGELOG.md:
```diff
@@ -1,3 +1,349 @@
+   LDK-based LSPs wishing to support often-offline senders and recipients should
+   set `UserConfig::enable_htlc_hold`, support the existing "onion mesage
+   mailbox" feature (setting `intercept_messages_for_offline_peers` on
+   `OnionMessegner` and handling `Event::OnionMessageIntercepted`s), and handle
```
Not sure if we also want to mention `Event::OnionMessagePeerConnected`

tnull commented on CHANGELOG.md:
```diff
@@ -1,3 +1,349 @@
+   * `KVStore` has been made async. Note that `KVStore` methods are not
+     `async fn`, but rather write ordering is fixed when the methods return,
+     though write completion is async.
+   * `BumpTransactionEventHandler` is no backed by an async `WalletSource` (and
```
`is no backed`?

tnull commented on CHANGELOG.md:
```diff
@@ -1,3 +1,349 @@
+ * The local key which receives non-HTLC-encumbered funds when the counterparty
+   force-closes a channel is now one of a static list of 1000 keys when using
+   `KeysManager` if `v2_remote_key_derivation` is set or after splicing (#4117).
+ * LSPS5 support was added, providing a push notification API for LSPS clients.
```
Link to the LSPS5 PR is missing.

TheBlueMatt replied:
I skipped links for features that had many PRs. In the case of LSPS5 there were at least 4/5 PRs with fixups so just left out links.

tnull replied:
Okay.

tnull commented on CHANGELOG.md:
```diff
@@ -1,3 +1,349 @@
+ * Repeated `Listen::block_disconnected` calls for each disconnected block in a
+   reorg have been replaced with a single `blocks_disconnected` call with the
+   fork point block (i.e. the highest block on both sides of the reorg, #3876).
+ * `lightning::routing::scoring::CombinedScorer` was added to better support
```
Could mention this allows to actually merge datasets?

joostjager replied:
Maybe that's too much detail? This changelog is pretty high level.

joostjager commented on CHANGELOG.md:
```diff
@@ -1,3 +1,154 @@
+
+## API Updates
+
+ * Splicing is now supported. The current implementation is expected to be
```
Curious why it says "expected". Isn't interop confirmed?

TheBlueMatt replied:
We have not had time to do interop testing, no. We will likely do some before the release but probably not a *ton*.

joostjager commented on CHANGELOG.md:
```diff
@@ -1,3 +1,154 @@
+   `UserConfig::reject_inbound_splices` can be set to block inbound ones (#4150)
+ * LDK now requires the `channel_type` feature in line with spec updates (#3896)
+
+XXX release stats
```
Reminder to update. Maybe XXX is part of the process?

TheBlueMatt replied:
Yea, will do it once we get the other PRs landed.

joostjager commented on CHANGELOG.md:
```diff
@@ -1,3 +1,154 @@
+# 0.2 - XXX, 2025 - "Natively Asynchronous Splicing"
```
Would it be nice to add a bit of a sales pitch here? Example AI slop:

LDK 0.2 – “Natively Asynchronous Splicing” is here, delivering the future of Lightning development today. Splicing support has landed, letting you resize channels on the fly, with native async APIs across the stack for faster, cleaner, and more efficient operations. Async payments are now live too, enabling smoother BOLT 12 flows for often-offline users and LSPs. Zero-Fee-Commitment channels slash feerate headaches, while new LSPS and RGS features power next-gen liquidity and performance. With major speed boosts, deeper diagnostics, and smarter persistence, LDK 0.2 is your all-in-one upgrade for a faster, more flexible Lightning future.

joostjager reviewed (commented):
I was assigned by the bot, but as @TheBlueMatt mentioned in the meet, everyone should really check their contributions here.

NIce concise changelogs

tankyleo commented on CHANGELOG.md:
```diff
@@ -1,3 +1,376 @@
+   mailbox" feature (setting `intercept_messages_for_offline_peers` on
+   `OnionMessenger` and handling `Event::OnionMessageIntercepted`s), and handle
+   `Event::PersistStaticInvoice`s and `Event::StaticInvoiceRequested`s.
+ * Zero-Fee-Commitment channels are now supported in LDK. These channels remove
```
Discussed offline, we should probably let people know these currently require the node and some of its peers to run Bitcoin Core v29+

tankyleo reviewed (commented):
This just crossed my mind: should we mention the new `max_tx_weight` parameter on `CoinSelectionSource::select_confirmed_utxos` in `API Updates` ? 

*`CoinSelectionSource::select_confirmed_utxos` now has a `max_tx_weight` parameter that specifies the maximum allowed weight of the final transaction (#4129). This parameter is only expected to put a real constraint on UTXO selection in zero-fee commitment channels."*

TheBlueMatt commented:
Backporting to 0.2 in #4193

TheBlueMatt commented:
Updated the blinded message path dummy hops note and moved it out of "bug fixes":

```diff
$ git diff-tree -U1 b2d9dacf81 19fd54f525
diff --git a/CHANGELOG.md b/CHANGELOG.md
index b4839d6423..513a026f62 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -92,2 +92,4 @@
    `KVStore` to restore custom logic for specific storage objects (#3905).
+ * `BlindedMessagePath::new_with_dummy_hops` was added (but is not used by
+   default, #3726). You can use `NodeIdMessageRouter` to enable dummy hops.
  * `ProbabilisticScoringFeeParameters::probing_diversity_penalty` was added to
@@ -150,4 +152,2 @@
    marginally when forwarding gossip to a slow peer (#4093, #4096).
- * `BlindedMessagePath::new_with_dummy_hops` was added (but is not used by
-   default, #3726).
  * Blinded path serialization is now padded to better hide its contents (#3177).
```

jkczyz commented on CHANGELOG.md:
```diff
@@ -1,3 +1,185 @@
+ * Upgrading to v0.2.0 will time out any pending async payment waiting for the
+   often offline peer to come online (#3918).
+ * Blinded message paths generated by previous versions of LDK, except those
+   generated for inclusion in `Bolt12Offer`s will no longer be accepted. As most
```
s/`Bolt12Offer`/BOLT12 `Offer`

TheBlueMatt commented:
Addressed further feedback:

```diff
$ git diff-tree -U2 19fd54f52 f36560d96
diff --git a/CHANGELOG.md b/CHANGELOG.md
index 513a026f62..517102954e 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -169,7 +169,7 @@
    often offline peer to come online (#3918).
  * Blinded message paths generated by previous versions of LDK, except those
-   generated for inclusion in `Bolt12Offer`s will no longer be accepted. As most
-   blinded message paths are ephemeral, this should only invalidate issued
-   `Refund`s in practice (#3917).
+   generated for inclusion in BOLT 12 `Offer`s will no longer be accepted. As
+   most blinded message paths are ephemeral, this should only invalidate issued
+   BOLT 12 `Refund`s in practice (#3917).
  * Once a channel has been spliced, LDK can no longer be downgraded.
    `UserConfig::reject_inbound_splices` can be set to block inbound ones (#4150)
diff --git a/lightning/src/ln/channelmanager.rs b/lightning/src/ln/channelmanager.rs
index 515d7dbd14..a77f086c53 100644
--- a/lightning/src/ln/channelmanager.rs
+++ b/lightning/src/ln/channelmanager.rs
@@ -4682,7 +4682,6 @@ where
 	/// [`Event::DiscardFunding`] is seen.
 	///
-	/// After initial signatures have been exchanged, if we contributed any inputs,
-	/// [`Event::FundingTransactionReadyForSigning`] will be generated and
-	/// [`ChannelManager::funding_transaction_signed`] should be called.
+	/// After initial signatures have been exchanged, [`Event::FundingTransactionReadyForSigning`]
+	/// will be generated and [`ChannelManager::funding_transaction_signed`] should be called.
 	///
 	/// If any failures occur while negotiating the funding transaction, an [`Event::SpliceFailed`]
```

tankyleo commented on CHANGELOG.md:
```diff
@@ -1,3 +1,387 @@
+   is implemented for all `Readable` (#3579).
+ * `LengthReadable` is now required to read various objects which consume the
+   full available buffer (#3640).
+ * structs in `lightning-liquidity` were renamed to be globally unique (#3583).
```
Structs
